src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -246,6 +275,83 @@ public virtual void WriteResponseHeaders(OutputFormatterWriteContext context)
         /// <returns>A task which can write the response body.</returns>
         public abstract Task WriteResponseBodyAsync(OutputFormatterWriteContext context);
 
+        /// <summary>
+        /// Adds or replaces the charset parameter in a given <paramref name="mediaType"/> with the
+        /// given <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="mediaType">The <see cref="StringSegment"/> with the media type.</param>
+        /// <param name="encoding">
+        /// The <see cref="Encoding"/> to add or replace in the <paramref name="mediaType"/>.
+        /// </param>
+        /// <returns>The mediaType with the given encoding.</returns>
+        protected StringSegment AddOrOverrideEncoding(StringSegment mediaType, Encoding encoding)
```

Comment:
I like more AddOrOverrideEncoding/Charset as it clearly implies what the method does, but will change it for GetMediaTypeWithCharset if you feel strongly about it.